File: tests/bindings/dotnet/foo.Tests/StringTest.cs
using System;
using System.Text;
using Xunit;
using foo;

namespace foo.Tests
{
    public class StringTest
    {
        const string ENGLISH_SENTENCE_1 = "I like to be home with my monkey and my dog";
        const string ENGLISH_SENTENCE_2 = "Don't care, shut up, play the record!";
        const string FRENCH_SENTENCE_1 = "Devant mon miroir j'ai rêvé d'être une star, j'ai rêvé d'être immortellement belle";
        const string FRENCH_SENTENCE_2 = "Ce soir j'irai voir à travers le miroir, si la vie est éternelle";

        [Fact]
        public void EnglishTest()
        {
            using(var stringclass = new StringClass())
            {
                Assert.Equal(ENGLISH_SENTENCE_1, stringclass.Echo(ENGLISH_SENTENCE_1));
                Assert.Equal(ENGLISH_SENTENCE_2, stringclass.Echo(ENGLISH_SENTENCE_2));
            }
        }

        [Fact]
        public void FrenchTest()
        {
            using (var stringclass = new StringClass())
            {
                Assert.Equal(FRENCH_SENTENCE_1, stringclass.Echo(FRENCH_SENTENCE_1));
                Assert.Equal(FRENCH_SENTENCE_2, stringclass.Echo(FRENCH_SENTENCE_2));
            }
        }

        [Fact]
        public void LengthTest()
        {
            Assert.Equal(Encoding.UTF8.GetByteCount(ENGLISH_SENTENCE_1), (int)StringClass.GetLength(ENGLISH_SENTENCE_1));
            Assert.Equal(Encoding.UTF8.GetByteCount(ENGLISH_SENTENCE_2), (int)StringClass.GetLength(ENGLISH_SENTENCE_2));
            // TODO: fix these with UTF8 support
            //Assert.Equal(Encoding.UTF8.GetByteCount(FRENCH_SENTENCE_1), (int)StringClass.GetLength(FRENCH_SENTENCE_1));
            //Assert.Equal(Encoding.UTF8.GetByteCount(FRENCH_SENTENCE_2), (int)StringClass.GetLength(FRENCH_SENTENCE_2));
        }

        [Fact]
        public void MemoryLeakTest()
        {
            const int NUM_ITERATIONS = 100000;

            using (var stringclass = new StringClass())
            {
                for (int i = 0; i < NUM_ITERATIONS; i++)
                {
                    Assert.Equal(FRENCH_SENTENCE_1, stringclass.Echo(FRENCH_SENTENCE_1));
                }
            }
        }
    }
}

File: tests/bindings/java/foo-tests/src/test/java/io/stepfunc/foo_test/StringTest.java
package io.stepfunc.foo_test;

import io.stepfunc.foo.StringClass;
import org.junit.jupiter.api.Test;

import java.io.UnsupportedEncodingException;
import java.nio.charset.StandardCharsets;

import static org.assertj.core.api.Assertions.assertThat;

public class StringTest {
    final String ENGLISH_SENTENCE_1 = "I like to be home with my monkey and my dog";
    final String ENGLISH_SENTENCE_2 = "Don't care, shut up, play the record!";
    final String FRENCH_SENTENCE_1 = "Devant mon miroir j'ai rêvé d'être une star, j'ai rêvé d'être immortellement belle";
    final String FRENCH_SENTENCE_2 = "Ce soir j'irai voir à travers le miroir, si la vie est éternelle";

    @Test
    public void EnglishTest() {
        try(StringClass stringclass = new StringClass()) {
            assertThat(stringclass.echo(ENGLISH_SENTENCE_1)).isEqualTo(ENGLISH_SENTENCE_1);
            assertThat(stringclass.echo(ENGLISH_SENTENCE_2)).isEqualTo(ENGLISH_SENTENCE_2);
        }
    }

    @Test
    public void FrenchTest() {
        try(StringClass stringclass = new StringClass()) {
            assertThat(stringclass.echo(FRENCH_SENTENCE_1)).isEqualTo(FRENCH_SENTENCE_1);
            assertThat(stringclass.echo(FRENCH_SENTENCE_2)).isEqualTo(FRENCH_SENTENCE_2);
        }
    }

    @Test
    public void LengthTest() throws UnsupportedEncodingException {
        assertThat(StringClass.getLength(ENGLISH_SENTENCE_1).intValue()).isEqualTo(ENGLISH_SENTENCE_1.getBytes(StandardCharsets.UTF_8).length);
        assertThat(StringClass.getLength(ENGLISH_SENTENCE_2).intValue()).isEqualTo(ENGLISH_SENTENCE_2.getBytes(StandardCharsets.UTF_8).length);
        assertThat(StringClass.getLength(FRENCH_SENTENCE_1).intValue()).isEqualTo(FRENCH_SENTENCE_1.getBytes(StandardCharsets.UTF_8).length);
        assertThat(StringClass.getLength(FRENCH_SENTENCE_2).intValue()).isEqualTo(FRENCH_SENTENCE_2.getBytes(StandardCharsets.UTF_8).length);
    }
}
